src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/actionfilter/FieldCapsFilter.kt

```diff
@@ -96,30 +125,63 @@ class FieldCapsFilter(
         }
     }
 
+    /**
+     * The FieldCapabilitiesResponse can contain merged or unmerged data. The response will hold unmerged data if its a cross cluster search.
+     *
+     * There is a boolean available in the FieldCapabilitiesRequest `isMergeResults` which indicates if the response is merged/unmerged.
+     * Unfortunately this is package private and when rewriting we can't access it from request. Instead will be relying on the response.
+     * If response has indexResponses then its unmerged else merged.
+     */
+    private fun rewriteResponse(response: FieldCapabilitiesResponse, rollupIndices: Set<String>, shouldDiscardResponse: Boolean): ActionResponse {
```

Comment:
I don't think we have any integ tests for ccs, can we add some unit tests for all this new logic?

Reply:
added unit tests